src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -170,6 +170,9 @@ public TracerSettings(IConfigurationSource source)
                                            "500-599";
             HttpServerErrorStatusCodes = ParseHttpCodesToArray(httpServerErrorStatusCodes);
 
+            Enum.TryParse(source?.GetString(ConfigurationKeys.Convention) ?? "default", ignoreCase: true, out ConventionType conventionType);
```

Comment:
As it is written, a setting of `"OpenTelemetry, Datadog"` will result in `(ConventionType)3`, which is invalid.

(`OpenTelemetry` is 1, `Datadog` is 2, and `TryParse()` will use bitwise `OR` even if the enum type doesn't have `[Flags]`)